Engine/ResourceManagement/Importer/ModelImporters/MeshImporter.cpp

```diff
@@ -27,25 +27,22 @@ void MeshImporter::ImportMesh(const aiMesh* mesh, const std::vector<std::string>
 		Mesh::Vertex new_vertex;
 		aiVector3D transformed_position = mesh_transformation * mesh->mVertices[i];
 		new_vertex.position = float3(transformed_position.x, transformed_position.y, transformed_position.z);
-		new_vertex.tex_coords = float2(mesh->mTextureCoords[0][i].x, mesh->mTextureCoords[0][i].y);
-		new_vertex.normals = float3(mesh->mNormals[i].x, mesh->mNormals[i].y, mesh->mNormals[i].z);
+		if (mesh->mTextureCoords[0]) 
+		{
+			new_vertex.tex_coords = float2(mesh->mTextureCoords[0][i].x, mesh->mTextureCoords[0][i].y);
+		}
+		if (mesh->mNormals)
+		{
+			new_vertex.normals = float3(mesh->mNormals[i].x, mesh->mNormals[i].y, mesh->mNormals[i].z);
+		}
 		vertices.push_back(new_vertex);
 	}
 
-	std::vector<uint32_t> materials_path_size;
-	uint32_t total_meshes_size = 0;
-	for (auto & path_size : loaded_meshes_materials)
-	{
-		materials_path_size.push_back(path_size.size());
-		total_meshes_size += path_size.size();
-	}
-
 	uint32_t num_indices = indices.size();
 	uint32_t num_vertices = vertices.size();
-	uint32_t num_materials = loaded_meshes_materials.size();
-	uint32_t ranges[3] = { num_indices, num_vertices, num_materials };
```

Comment:
If you don't save the materials here, then you force the engine users to assign each texture again each time the model is used.

Reply:
I'm saving it here https://github.com/Unnamed-Company/LittleOrionEngine/pull/4/files#diff-72a87ab716cc5de90d9eec39c3e5f3a3R118 now

Reply:
And the structure I can't do it like that, because there are meshes that share the same skeleton, for example the zombunny. Ctrl_Grp is the name inside the fbx, so we can tell the artists to name the skeleton as skeleton.